platform/android/src/native_map_view.cpp

```diff
@@ -127,11 +127,6 @@ std::array<uint16_t, 2> NativeMapView::getFramebufferSize() const {
 void NativeMapView::activate() {
     mbgl::Log::Debug(mbgl::Event::Android, "NativeMapView::activate");
 
-    oldDisplay = eglGetCurrentDisplay();
```

Comment:
This doesn't look correct to me -- we want `deactivate` to restore the display and surfaces that were active when `activate` was called. We shouldn't assume that's the same display and surfaces that were active when `createSurface` was called.

I think rather what we should do is call `activate` / `deactivate` _inside_ `createSurface`, around the portion of the code where we want our context active.